test/integration/patients/patient/dashboard.js

```diff
@@ -380,189 +403,254 @@ context('patient dashboard page', function() {
   });
 
   specify('add action and flow', function() {
+    const currentClinican = getCurrentClinician({
+      relationships: {
+        role: getRelationship(roleEmployee),
+      },
+    });
+
     cy
       .routeCurrentClinician(fx => {
-        fx.data.relationships.role = { data: { id: '33333' } };
+        fx.data = currentClinican;
+
         return fx;
       })
       .routesForPatientAction()
       .routePatient(fx => {
-        fx.data.id = '1';
+        fx.data = getPatient({
+          id: '1',
+        });
 
         return fx;
       })
       .routePrograms(fx => {
-        fx.data = _.sample(fx.data, 5);
-
-        fx.data[0].id = 1;
-        fx.data[0].attributes.name = 'Two Actions, One Published, One Flow';
-        fx.data[0].attributes.published_at = testTs();
-        fx.data[0].attributes.archived_at = null;
-        fx.data[0].relationships['program-actions'] = {
-          data: [
-            { id: '1' },
-            { id: '4' },
-            { id: '5' },
-            { id: '6' },
-          ],
-        };
-        fx.data[0].relationships['program-flows'] = { data: [{ id: '4' }] };
-
-        fx.data[1].id = 2;
-        fx.data[1].attributes.name = 'Two Published Actions and Flows';
-        fx.data[1].attributes.published_at = testTs();
-        fx.data[1].attributes.archived_at = null;
-        fx.data[1].relationships['program-actions'] = {
-          data: [
-            { id: '2' },
-            { id: '3' },
-            { id: '4' },
-            { id: '5' },
-            { id: '6' },
-          ],
-        };
-        fx.data[1].relationships['program-flows'] = {
-          data: [
-            { id: '5' },
-            { id: '6' },
-            { id: '7' },
-            { id: '8' },
-            { id: '9' },
-          ],
-        };
-
-        fx.data[2].id = 3;
-        fx.data[2].attributes.name = 'No Actions, No Flows';
-        fx.data[2].attributes.published_at = testTs();
-        fx.data[2].attributes.archived_at = null;
-        fx.data[2].relationships['program-actions'] = { data: [] };
-        fx.data[2].relationships['program-flows'] = { data: [] };
-
-        fx.data[3].id = 4;
-        fx.data[3].attributes.name = 'Should not show - unpublished';
-        fx.data[3].attributes.published_at = null;
-        fx.data[3].attributes.archived_at = null;
-        fx.data[3].relationships['program-actions'] = { data: [] };
-        fx.data[3].relationships['program-flows'] = { data: [] };
-
-        fx.data[4].id = 5;
-        fx.data[4].attributes.name = 'Should not show - archived';
-        fx.data[4].attributes.published_at = testTs();
-        fx.data[4].attributes.archived_at = testTs();
-        fx.data[4].relationships['program-actions'] = { data: [] };
-        fx.data[4].relationships['program-flows'] = { data: [] };
+        fx.data = [
```

Comment:
this test setup would be best to create the elements at the top using the uuids that come back.. _however_ there is a circular relationship so some of the `getFoo`s will need to be passed `uuid`s since they need to be used in a relationship prior to their instantiation.

Reply:
I made some updates to this test. But probably more that can be done here.